### DISCUSSION TITLE
Allow for storing and loading Architectures as YAML

### DISCUSSION THREAD

**Discussion by Raahul-Singh** [Lightning-AI/pytorch-lightning#10464] [Idea pool]
Hey folks, I often use [@ferrine's](https://gist.github.com/ferrine) [network builder ](https://gist.github.com/ferrine/89d739e80712f5549e44b2c2435979ef)to store my architectures as yaml files.
It is very convenient for complicated architectures and custom models. 
You reckon we can introduce this here?
Probably store all the necessary config for a lightning module in a YAML and build the object through a factory class.

**Comment by ananthsub** [comment]
I strongly believe the Lightning framework should not require a configuration system for its use. The Python API is the most expressive. 

That being said, if you find that system convenient to write your projects, feel free to use it! I believe the user should determine the config choice (if any) that works best for them

**Reply by Raahul-Singh** [reply]
Sounds fair!
